cmd/sidecar/main.go

```diff
@@ -0,0 +1,233 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+	"github.com/percona/percona-server-mysql-operator/pkg/mysql"
+	"github.com/pkg/errors"
+)
+
+var log = logf.Log.WithName("sidecar")
+
+type BackupConf struct {
+	Destination string                        `json:"destination"`
+	Storage     apiv1alpha1.BackupStorageSpec `json:"storage"`
+}
+
+func main() {
+	opts := zap.Options{Development: true}
+	logf.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/health/", func(w http.ResponseWriter, req *http.Request) {
+		fmt.Fprintf(w, "OK")
+	})
+	mux.HandleFunc("/backup/", backupHandler)
+	mux.HandleFunc("/logs/", logHandler)
+
+	log.Info("starting http server")
+	log.Error(http.ListenAndServe(":6033", mux), "http server failed")
+}
+
+func getNamespace() (string, error) {
+	ns, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return "", errors.Wrap(err, "read namespace file")
+	}
+
+	return string(ns), nil
+}
+
+func xtrabackupArgs() []string {
+	return []string{
+		"--backup",
+		"--stream=xbstream",
+		fmt.Sprintf("--user=%s", os.Getenv("BACKUP_USER")),
```

Comment:
We need to use all secrets not  from  env vars but from files. For these needs we need to mount  secret to the fs (we use the same approach in pxc operator e.g. monitor user for haproxy ). Why is it needed? It is needed for secrets rotation. If I need to change AWS secrets the mysql pods should not be restarted.  We can do it via separate task for next release but please create the task.

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/77/commits/452740c192e52d151fcf01b42b6ff732f92aa077